service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureControlledStorageResourceServiceTest.java

```diff
@@ -0,0 +1,273 @@
+              null,
+              storageAccount.getResourceId(),
+              "sc-" + storageContainerId);
+      controlledResourceService.createControlledResourceSync(
+          storageContainer,
+          null,
+          user.getAuthenticatedRequest(),
+          getAzureStorageContainerCreationParameters());
+    }
+    return AzureControlledStorageResourceServiceTest.reusableWorkspace;
+  }
+
+  private void resetSecondUserWorkspaceAccess(UUID workspaceUuid) throws Exception {
+    List<RoleBinding> roles =
+        SamRethrow.onInterrupted(
+            () ->
+                samService.listRoleBindings(
+                    workspaceUuid, workspaceOwner.getAuthenticatedRequest()),
+            "listRoleBindings");
+    for (RoleBinding role : roles) {
+      if (role.users().contains(userAccessUtils.getSecondUserEmail())) {
+        workspaceService.removeWorkspaceRoleFromUser(
+            workspaceUuid,
+            role.role(),
+            userAccessUtils.getSecondUserEmail(),
+            workspaceOwner.getAuthenticatedRequest());
+      }
+    }
+  }
+
+  private void assertValidToken(String sas, BlobContainerSasPermission expectedPermissions) {
+    Pattern protocolRegex = Pattern.compile("spr=https&");
+    // SAS tokens start and expiry times are UTC
+    Pattern startTimeRegex =
+        Pattern.compile(
+            "st="
+                + startTime
+                    .atZoneSameInstant(ZoneOffset.UTC)
+                    .format(DateTimeFormatter.ISO_LOCAL_DATE));
+    Pattern expiryTimeRegex =
+        Pattern.compile(
+            "se="
+                + expiryTime
+                    .atZoneSameInstant(ZoneOffset.UTC)
+                    .format(DateTimeFormatter.ISO_LOCAL_DATE));
+    Pattern signedResourceRegex = Pattern.compile("sr=c&");
+    Pattern permissionsRegex = Pattern.compile("sp=" + expectedPermissions.toString() + "&");
+
+    assertThat("SAS is https", protocolRegex.matcher(sas).find());
+    assertThat("SAS validity starts today", startTimeRegex.matcher(sas).find());
+    assertThat("SAS validity ends today", expiryTimeRegex.matcher(sas).find());
+    assertThat("SAS is for a container resource", signedResourceRegex.matcher(sas).find());
+    assertThat("SAS grants correct permissions", permissionsRegex.matcher(sas).find());
+  }
+
+  /**
+   * Set up default values for user, workspace for tests to use. By default, this will point to the
+   * reusable workspace created by {@code reusableWorkspace}.
+   */
+  @BeforeEach
+  public void setupUserAndWorkspace() throws Exception {
+    workspaceOwner = userAccessUtils.defaultUser();
+    workspace = reusableWorkspace(workspaceOwner);
+    resetSecondUserWorkspaceAccess(workspace.getWorkspaceId());
+  }
+
+  /**
+   * Reset the {@link FlightDebugInfo} on the {@link JobService} to not interfere with other tests.
+   */
+  @AfterEach
+  public void resetFlightDebugInfo() {
+    // Exercise enumeration by dumping after each
```

Comment:
I don't understand lines 207-209. Do you have any details about why we need to so these after every test?